taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/taxonomy/storage/JdbcTemplateEx.java

```diff
@@ -0,0 +1,87 @@
+package com.digirati.taxman.rest.server.taxonomy.storage;
+
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.RowMapperResultSetExtractor;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.util.CollectionUtils;
+
+import javax.sql.DataSource;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * An extension of JdbcTemplate that provides a queryForOptional, as a variant
+ * of queryForObject that doesn't throw if the queried object is not found.
+ */
+public class JdbcTemplateEx extends JdbcTemplate {
```

Comment:
We don't need this implemented for `Optional` with every overload. A better solution is to just create a single static helper that can go from `List<T>` to `Optional<T>`.

Reply:
That omits having to add another statement to each call, kind of keeps with the current flow&design. Also if ever JdbcTemplate gets such method, this can be very easily retired.

Reply:
I agree that keeping the succinct code style in the various `XDao` classes is a good goal, but subclassing `JdbcTemplate` isn't the way to do it.